src/Planners/src/CubicSpline.cpp

```diff
@@ -0,0 +1,768 @@
+    void addTripletCurrentKnot(const int& knotIndex,
+                               const int& rowOffset,
+                               const int& columnOffset,
+                               std::vector<Eigen::Triplet<double>>& tripletList);
+
+    /**
+     * addTripletPreviousKnot is a helper function to generate a triplet containing a 2x2 matrix.
+     * This matrix is used to compute the intermediate velocity and acceleration.
+     */
+    void addTripletPreviousKnot(const int& knotIndex,
+                                const int& rowOffset,
+                                const int& columnOffset,
+                                std::vector<Eigen::Triplet<double>>& tripletList);
+    /**
+     * addTripletNextKnot is a helper function to generate a triplet containing a 2x2 matrix.
+     * This matrix is used to compute the intermediate velocity and acceleration.
+     */
+    void addTripletNextKnot(const int& knotIndex,
+                            const int& rowOffset,
+                            const int& columnOffset,
+                            std::vector<Eigen::Triplet<double>>& tripletList);
+
+    /**
+     * addKnownTermKnotPosition is a helper function to generate a 2-d vector,
+     * This vector is the known term used to compute the intermediate velocity and acceleration.
+     */
+    void addKnownTermKnotPosition(const std::size_t& knotIndex,
+                                  const std::size_t& coordinateIndex,
+                                  Eigen::Ref<Eigen::VectorXd> b);
+
+    /**
+     * addKnownTermNextKnot is a helper function to generate a 2-d vector,
+     * This vector is the known term used to compute the intermediate velocity and acceleration.
+     */
+    void addKnownTermNextKnot(const std::size_t& knotIndex,
+                              const std::size_t& coordinateIndex,
+                              Eigen::Ref<Eigen::VectorXd> b);
+
+    /**
+     * addKnownTermPreviousKnot is a helper function to generate a 2-d vector,
+     * This vector is the known term used to compute the intermediate velocity and acceleration.
+     */
+    void addKnownTermPreviousKnot(const std::size_t& knotIndex,
+                                  const std::size_t& coordinateIndex,
+                                  Eigen::Ref<Eigen::VectorXd> b);
+
+    /**
+     * Compute the intermediate velocities
+     */
+    void computeIntermediateVelocities();
+
+    /**
+     * Function to set the coefficients of the polynomials
+     */
+    void setPolynomialCoefficients(Polynomial& poly);
+
+    /**
+     * This function is the entry point to compute the coefficients of the spline
+     */
+    bool computeCoefficients();
+};
+
+CubicSpline::CubicSpline()
+{
+    m_pimpl = std::make_unique<Impl>();
+}
+
+CubicSpline::~CubicSpline() = default;
+
+bool CubicSpline::setInitialConditions(Eigen::Ref<const Eigen::VectorXd> velocity,
+                                       Eigen::Ref<const Eigen::VectorXd> /**acceleration*/)
```

Comment:
Not sure whether the second parameter is left on purpose here. If so, how is it exploited?

Reply:
`acceleration` is not used however is required to keep the same interface between the two splines